### ISSUE TITLE
Download status is not continued after resume

### ISSUE THREAD

**Issue by JakobDev** [Heroic-Games-Launcher/HeroicGamesLauncher#417]
I currently have the problem that the download status is not continued after resume a download. According to the [1.7.0 release notes](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/releases/tag/v1.7.0):
> Heroic now will continue the download status from where it stopped instead of getting back from 0% every time.

But that does not work for me. The download still starts at 0% and 0MiB after I restart the download of a game after closing the launcher. There are new files appearing in the game folder so I think the problem is just the download status. I'm using Manjaro with Heroic installed from AUR.

**Comment by flavioislima** [comment]
Yeah, this feature will work only if you manually stop the download and click on keep files. Not if you close the launcher right now.

**Comment by JakobDev** [comment]
Any chance this will be fixed in the near future?

**Comment by flavioislima** [comment]
might be, should not be a complex task since we already give a warning about closing while downloading.

**Comment by flavioislima** [comment]
Unfortunately that's not how it works.
Legendary download the games and when it restarts a download, it starts from zero percent from the remaining files.

The workaround we have in heroic, involves storing the percent when you click cancel and keep files.
Then, when you click continue, we do some calculations based on the amount we stored and what legendary send to us.

So it's not like heroic knows from beginning, we need a workaround for that.
If legendary sent the right percent, we wouldn't need this. Maybe they change this in the future but I don't know. 

So it's possible to keep the download status when closing heroic. But if the process get killed, heroic will never have the time to store the percent and will start from zero.

**Comment by flavioislima** [comment]
> Whats with writing the percent to a file everytime it's changed?

We are already writting the install log. Yeah, maybe we could parse the log before restart, getting the latest percent. Might be a good idea. 

But we have plans on stopping to write this log. But still, could be possible to write the percent on the app storage. 🤔

Yeah, I don't know, I'll make some tests when I have some time.

**Comment by adityaruplaha** [comment]
> So it's not like heroic knows from beginning, we need a workaround for that.
If legendary sent the right percent, we wouldn't need this. Maybe they change this in the future but I don't know.

I don't think they should actually. When I see 0%, I automatically assume 0% of what's left. Much like how pacman resumes downloads. Never was a fan of percentages resuming.

**Comment by JakobDev** [comment]
> > So it's not like heroic knows from beginning, we need a workaround for that.
> > If legendary sent the right percent, we wouldn't need this. Maybe they change this in the future but I don't know.
> 
> I don't think they should actually. When I see 0%, I automatically assume 0% of what's left. Much like how pacman resumes downloads. Never was a fan of percentages resuming.

The most game launchers save the progress, so any new user of Heroic will think that the download is complete started from beginning instead of downloading the rest. maybe a message box could help here.

A option to toggle the behaviour between showing saving the progress and show how much is left would be good, so everyone can choose what he prefers.

**Comment by Ryder17z** [comment]
percentages should be of the total in my opinion.
